### ISSUE TITLE
yaml formatting/linting

### ISSUE THREAD

**Issue by jakkdl** [python-trio/trio#3164]
breaking out from #3157 

in flake8-async we've been using https://github.com/jumanjihouse/pre-commit-hook-yamlfmt for a while, but it mostly just handles indentation.

Looking around a bit I found yamllint, which has rules for quoting - with sufficient complexity it should satisfy @webknjaz (https://github.com/python-trio/trio/pull/3157#discussion_r1891024219)
https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.quoted_strings

Neither of the above tools can enforce block- vs flow-style arrays, but I found https://lyz-code.github.io/yamlfix/#sequence-list-style

yamlfix also has some quote rules: https://lyz-code.github.io/yamlfix/#quote-basic-values

tl;dr: yamlfix should be good

**Comment by webknjaz** [comment]
Yeah, I tend to use yamllint (it can be integrated into pre-commit natively). But I don't normally change many defaults. Typically, I just set the sequences not to have extra indents. The quotes toggle seems interesting, but I haven't tried it out myself.

**Comment by webknjaz** [comment]
> in flake8-async we've been using [jumanjihouse/pre-commit-hook-yamlfmt](https://github.com/jumanjihouse/pre-commit-hook-yamlfmt?rgh-link-date=2024-12-22T11%3A40%3A28Z) for a while, but it mostly just handles indentation.

Oh, I didn't realize it works with the `.yamllint` config..

**Comment by kachida** [comment]
@jakkdl , I'm interested and could you kindly assign me this task.

**Comment by A5rocks** [comment]
Done though you can always just make a PR regardless!
